recipes/openblas/all/conanfile.py

```diff
@@ -19,134 +23,221 @@ class OpenblasConan(ConanFile):
         "fPIC": [True, False],
         "build_lapack": [True, False],
         "use_thread": [True, False],
+        "use_openmp": [True, False],
         "dynamic_arch": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "build_lapack": False,
```

Comment:
I'd suggest if it's possible to build lapack without fortran (I thought this was just a C api that ultimately still needed to link to the fortran libs?), then we should build it by default.  

Because of the requirement that all CCI packages must build with the defaults of their dependencies, this opens the door to a number of scientific packages that choke without lapack support.

```suggestion
        "build_lapack": True,
```

Reply:
Since openblas v0.3.21, it is possible to build with LAPACK support without Fortran. A f2c (fortran to c) converted copy of LAPACK 3.9.0 is bundled, which completely avoids the need for a Fortran runtime, see https://github.com/xianyi/OpenBLAS/releases/tag/v0.3.21 why 3.9.0 and not newer. This allows us to provide LAPACK on CCI without a Fortran compiler. One related issues is #4509.

Reply:
I have added a check in `config_options()` that changes the default `build_lapack` to `True` if the version is new enough. Note that the recipe then will have different default options depending on package version. I do not know if this against CCI policy.

Reply:
If that's the case, I'm not sure it's desirable to introduce different defaults across versions. My intuition would be to keep it stable across versions (so keep `"build_lapack": False`, but this has to be balanced with the benefits of having lapack available to support other conan center packages. @uilianries @prince-chrismc can either of you weigh in here?

Reply:
Reminder to possibly weigh in regarding this suggested change. @uilianries @prince-chrismc.

Reply:
This has been scheduled for discussion with the team, thanks a lot for your patience

Reply:
Any update on this topic? I'm willing to forego setting `"build_lapack": True` for >=0.3.21, alternatively, build both `[True, False]` (if possible) if this could expedite the merging of this pull request.